static/js/compose_validate.js

```diff
@@ -654,6 +655,81 @@ function validate_private_message() {
     return true;
 }
 
+export function check_send_permissions() {
+    const send_button = document.querySelector("#compose-send-button");
+    const send_button_container = document.querySelector(".compose_right_float_container");
+    const tippy_instance = send_button_container._tippy || tippy(send_button_container);
+    const msg_type = compose_state.get_message_type();
+
+    if (msg_type === "private" && page_params.realm_private_message_policy === 2) {
+        const recipients = compose_pm_pill.get_user_ids();
+
+        const non_bot_users = recipients.filter(
+            (user_id) => !people.get_by_user_id(user_id).is_bot,
+        );
+
+        if (non_bot_users.length > 0) {
+            tippy_instance.setContent(
+                $t_html({
+                    defaultMessage:
+                        "You are not allowed to send private messages in this organization.",
+                }),
+            );
+            send_button.classList.add("button_disabled");
```

Comment:
Do we actually use this class? I cannot find it defining any CSS or anything else.

Reply:
Thanks for pointing this out, there is no longer a reason for this class and the reference to it is now removed in the latest commit